### ISSUE TITLE
Pytest hangs forever with Github Actions

### ISSUE THREAD

**Issue by Mulugruntz** [pytest-dev/pytest#8700]
After collecting the items, it hangs until Github's timeout.
Example there: https://github.com/Mulugruntz/celery-pubsub/runs/2668823507?check_suite_focus=true#step:7:22

What is weird is that on another project, setup in a very similar (but not same) way, it works fine:
https://github.com/Mulugruntz/aiosubprocess/runs/2665341529?check_suite_focus=true#step:7:50

On my local, it works (famous words... 🙄 )

**on local**
<details>

```
coverage run -m pytest tests && coverage report && coverage xml
========================================= test session starts ==========================================
platform darwin -- Python 2.7.16, pytest-4.6.11, py-1.10.0, pluggy-0.13.1
rootdir: /Users/sgiffard/PycharmProjects/celery-pubsub, inifile: pytest.ini, testpaths: tests
collected 10 items                                                                                     

tests/test_pubsub.py ..........                                                                  [100%]

====================================== slowest 10 test durations =======================================
0.01s call     tests/test_pubsub.py::test_subscription

(0.00 durations hidden.  Use -vv to show these durations.)
====================================== 10 passed in 0.10 seconds =======================================
Name                        Stmts   Miss Branch BrPart  Cover
-------------------------------------------------------------
celery_pubsub/__init__.py       2      0      0      0   100%
celery_pubsub/pubsub.py        47      0     10      0   100%
tests/__init__.py               0      0      0      0   100%
tests/conftest.py              49      0      0      0   100%
tests/test_pubsub.py           62      0      0      0   100%
-------------------------------------------------------------
TOTAL                         160      0     10      0   100%

```

</details>

**on Github Actions (first link, above)**

<details>

```
============================= test session starts ==============================
platform linux2 -- Python 2.7.18, pytest-4.6.11, py-1.10.0, pluggy-0.13.1
rootdir: /home/runner/work/celery-pubsub/celery-pubsub, inifile: pytest.ini, testpaths: tests
plugins: celery-4.4.7
collected 10 items
```

</details>

Only Python version is different (but minor). And it's on python 2.7 for the example but it also fails on all the other tested versions (cpython 2.7, 3.6, 3.7, 3.8 and pypy 2.7, 3.6, 3.7).

It looks like a bug. Not sure where. Maybe Github Actions, maybe pytest?

**Comment by RonnyPfannschmidt** [comment]
at first glance a difference seems to be the celery plugin,
a next suggestion is to experiment with timeouts and to run `pytest -s` to disable output capture

**Comment by Mulugruntz** [comment]
Thanks @RonnyPfannschmidt for your quick reply!

I'm trying right now with `-s`: https://github.com/Mulugruntz/celery-pubsub/runs/2669055301?check_suite_focus=true
So far, it seems to be the same.

The test suite runs in less than a second, not sure why timeout would be needed.

I'll also try with `-v`. Not sure if it'll help 🤞

**Comment by Mulugruntz** [comment]
It seems to just hang after collecting the items and before running the suite (verbose gives nothing new)
https://github.com/Mulugruntz/celery-pubsub/runs/2669090934?check_suite_focus=true#step:7:23

**Comment by Mulugruntz** [comment]
I tried by adding `pytest-celery` but it's still the same 
https://github.com/Mulugruntz/celery-pubsub/runs/2669205172?check_suite_focus=true#step:7:25

From my understanding, it just forces `celery>=4.4.0` to use `celery.comtrib.pytest`. But nothing changed.

Time to sleep, it's super late (early) here.

**Comment by Mulugruntz** [comment]
You're right.
From Celery 4.x, not only `CELERY_ALWAYS_EAGER=True` is no longer recommended but also it doesn't work anymore. 
source: https://docs.celeryproject.org/en/stable/userguide/testing.html

Therefore, it works in Celery 3.x but fails in Celery 4.x and 5.x.
Pretty bad that it hangs with no message, no logging, nothing. I'll open a ticket on Celery side to request some help. I've tried for many hours to make it work in 4.x with their fixtures, to no avail.

Example in Celery 3.x: https://github.com/Mulugruntz/celery-pubsub/actions/runs/878351415

So, I'll close the ticket here, but of course feel free to reopen it if you think of some solution to make Celery 4.x work with pytest in my particular case.

**Comment by em1208** [comment]
@Mulugruntz I'm having the same issue with pytest, currently running `celery==4.4.2`. Pytest hangs only on github action and I was not able to figure out the issue. On the testing environment `CELERY_ALWAYS_EAGER` is not enabled so it should not matter. Where you able to figure out what's the cause? I would really appreciate your reply!

**Comment by Mulugruntz** [comment]
@em1208 No, I had some other stuffs to take care of. I haven't been able to fix it yet.

**Comment by kahkeong** [comment]
I also faced the same issue but I found out the root cause is due to Django version. I was using Django 3.1.12 with two databases aliases like below.

```
DATABASES = {
    "default": env.db("DATABASE_URL"),
    "another_connection": env.db("DATABASE_URL")
}
DATABASES["another_connection"]['OPTIONS'] = {'isolation_level': psycopg2.extensions.ISOLATION_LEVEL_REPEATABLE_READ}
```
Occasionally, Django will use the second connection to create the test database and I believe it messed up the test setup and caused my Github Actions to stuck after "Collected xx items"
This bug is resolved at https://github.com/django/django/pull/13507/files with Django 3.2 version, haven't encountered this issue since I upgraded Django to 3.2. Hope you find this comment helpful !
